### DISCUSSION TITLE
Creating related objects upon creation of object

### DISCUSSION THREAD

**Discussion by lamwj98** [laravel/nova-issues#5977] [Q&A]
My project is a little complex and is used to create an operational administration panel for non-tech personnels.

I have a Product model which have prices, images and other models that depend on this Product model. By user experience, they want to be able to add everything in one creation form so I need to create a form such that it can create multiple models at the same time. What I can think of right now is to use a custom logic to create the separate models.

In this case, prices, images and other models have a foreign key which points to product. I also want to be able to create multiple images model at the same time and point them to product. Wondering if there is any plan to implement such a feature.

**Comment by davidhemphill** [comment]
This is already possible by using the `fillUsing` method on a field. Take note if you return a function in that method it will be called after all the other fields have filled the resource, giving you a perfect place to create relations. This is actually how the `Repeater` field presets work, so take a look there for inspiration on how to implement this.

https://nova.laravel.com/docs/resources/fields.html#field-hydration
